lib/elixir/lib/calendar/iso.ex

```diff
@@ -258,4 +460,24 @@ defmodule Calendar.ISO do
       _ -> :error
     end
   end
+
+  def valid_date?(year, month, day)
+  def valid_date?(year, month, day) when month in (1..12) and day in (1..31) do
```

Comment:
We don't need the guards. The month and day ranges are checked by `:calendar` already.